lib/segment/src/index/field_index/map_index/mmap_map_index.rs

```diff
@@ -247,8 +247,15 @@ impl<N: MapIndexKey + Key + ?Sized> MmapMapIndex<N> {
     }
 
     pub fn iter_values_map(&self) -> impl Iterator<Item = (&N, IdIter<'_>)> + '_ {
-        self.value_to_points
-            .iter()
-            .map(|(k, v)| (k, Box::new(v.iter().copied()) as IdIter))
+        self.value_to_points.iter().map(|(k, v)| {
+            (
+                k,
+                Box::new(
+                    v.iter()
+                        .copied()
+                        .filter(|idx| !self.deleted.get(*idx as usize).unwrap_or(true)),
```

Comment:
Shouldn't it be unwrap or false? If our deleted bitvec is not big enough, we assume the point is not deleted yet.

```suggestion
                        .filter(|idx| !self.deleted.get(*idx as usize).unwrap_or(false)),
```

Reply:
It is not possible that the deleted slice is shorter, since it gets created at the specific length and never resizes. So if we check for a larger idx, it should mean it is not there

Reply:
But, it'll also never be longer, right? And we control what points we query from it.

Maybe 

```rust
!self.deleted.get[*idx as usize]
```

will suffice.

Reply:
I don't trust direct indexing at this point 😅

Reply:
Okay! 😆